sopn_publish_date/__init__.py

```diff
@@ -19,23 +23,50 @@ def __init__(self):
             "gla": self.greater_london_assembly,
             "pcc": self.police_and_crime_commissioner,
             "mayor": self.mayor,
-            "parl": self.uk_parliament,
         }
         self.calendar = UnitedKingdomBankHolidays()
 
-    def for_id(self, election_id: str) -> date:
+    def for_id(self, election_id: str, country: str = None) -> date:
         """
-        Calculate the publish date for an election given in `uk-election-ids <https://elections.democracyclub.org.uk/reference_definition/>`_ format, or raise an exception if that election id is ambiguous (could correspond to elections in multiple countries with different electoral legislation)
+        Calculate the publish date for an election given in `uk-election-ids <https://elections.democracyclub.org.uk/reference_definition/>`_ format and an optional country if necessary (for example, local or parliamentary elections), or raise an exception if that election id is ambiguous (could correspond to elections in multiple countries with different electoral legislation)
 
         :param election_id: a string representing an election id in uk-election-ids format
+        :param country: an optional string representing the country where the election will be held
         :return: a datetime representing the expected publish date
         """
+
         election_type, poll_date = type_and_poll_date(election_id)
 
+        countries = {
+            "ENG": Country.ENGLAND,
+            "WLS": Country.WALES,
+            "SCT": Country.SCOTLAND,
+            "NIR": Country.NORTHERN_IRELAND,
+        }
+
+        def valid_election_type(el_type):
+            return el_type in self.election_id_lookup or el_type in ["local", "parl"]
```

Comment:
Debatable if we need it ;) but I think `europarl` was also a different SOPN date for different places?

Reply:
Hah, given that there's no 1:1 mapping of Country -> Region within England, and also the very obvious other reason, I've taken the executive decision that `for_id` returns `None` for `europarl` and will be documented accordingly.